### PR TITLE
imxrt-multi: add optional libdummyfs support

### PR DESCRIPTION
**Pull request by gerard5**
Depending on board config `imxrt-multi` can be build with incorporated `dummyfs` support. Implementation is based on `stm32l4-multi/fs.c`, but does not assume that `libdummyfs` is always compiled with `imxrt-multi`.

Implementer may choose if he/she want to use classic `dummyfs server` or `imxrt-multi with libdummyfs`, depending on project demands and how memory is divided on the target, how booting is done: whether from xip or ram.

To build `imxrt-multi` with `libdummyfs` the project maintainer need to define in board config:
[_projects/armv7m7-imxrt117x-yourprojectname/board_config.h](https://github.com/phoenix-rtos/phoenix-rtos-project/blob/master/_projects/armv7m7-imxrt117x-evk/board_config.h)
[_projects/armv7m7-imxrt106x-yourprojectname/board_config.h](https://github.com/phoenix-rtos/phoenix-rtos-project/blob/master/_projects/armv7m7-imxrt106x-evk/board_config.h)
the following:
`#define BUILTIN_DUMMYFS 1`

Requires:
https://github.com/phoenix-rtos/phoenix-rtos-filesystems/pull/98

JIRA: NIL-432

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: nil-imxrt1176

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
https://github.com/phoenix-rtos/phoenix-rtos-filesystems/pull/98
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: multi/imxrt-multi/fs.c [hunk 1]
```diff
@@ -0,0 +1,170 @@
+/*
+ * Phoenix-RTOS
+ *
+ * STM32L4 Filesystem driver
```
Comment (agkaminski) [inline multi/imxrt-multi/fs.c:4]:
```suggestion
 * i.MX RT Filesystem driver
```

File: multi/imxrt-multi/imxrt-multi.c [hunk 2]
```diff
@@ -207,9 +206,11 @@ static int createDevFiles(void)
...
 	char name[8];
 
-	while (lookup("/", NULL, &dir) < 0) {
-		usleep(100000);
-	}
+#if (BUILTIN_DUMMYFS) == 0
+	/* Wait for the filesystem */
+	while (lookup("/", NULL, &dir) < 0)
+		usleep(10000);
+#endif
```
Comment (agkaminski) [inline multi/imxrt-multi/imxrt-multi.c:213]: Uh, placement of this waiting loop is and was buggy, we create a file in `main()` before invoking `createDevFiles()`. Maybe move this to `main()`?

File: multi/imxrt-multi/fs.c [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <board_config.h>
+#include "config.h"
+
+#if BUILTIN_DUMMYFS
```
Comment (agkaminski) [inline multi/imxrt-multi/fs.c:17]: Perhaps add this `#if` to STM32 too?

File: multi/imxrt-multi/imxrt-multi.c (hunk 2 above)
Comment (gerard5) [reply multi/imxrt-multi/imxrt-multi.c:213]: Yes, exactly - I think so too. I'll move it to the right place.

File: multi/imxrt-multi/fs.c (hunk 3 above)
Comment (gerard5) [reply multi/imxrt-multi/fs.c:17]: Done.

File: multi/imxrt-multi/fs.c (hunk 1 above)
Comment (gerard5) [reply multi/imxrt-multi/fs.c]: Done.

**Review by agkaminski — APPROVED** [review]
LGTM

**Pull request merged**